libraries/c_sdk/standard/ble/utest/iot_ble_data_transfer_utest.c

```diff
@@ -0,0 +1,1495 @@
+            TEST_ASSERT( cb.pConnectionCb );
+            lot_service.connection_callback = cb.pConnectionCb;
+            break;
+
+        case eBLEMtuChanged:
+            TEST_ASSERT( cb.pMtuChangedCb );
+            lot_service.mtu_changed_callback = cb.pMtuChangedCb;
+            break;
+
+        default:
+            /* Don't want to assert. New event callbacks may be added in the future */
+            return eBTStatusFail;
+    }
+
+    return eBTStatusSuccess;
+}
+
+
+static BTStatus_t IotBle_UnregisterEventCb_Callback( IotBleEvents_t event,
+                                                     IotBleEventsCallbacks_t cb,
+                                                     int n_calls )
+{
+    switch( event )
+    {
+        case eBLEConnection:
+            TEST_ASSERT( cb.pConnectionCb );
+            lot_service.connection_callback = NULL;
+            break;
+
+        case eBLEMtuChanged:
+            TEST_ASSERT( cb.pMtuChangedCb );
+            lot_service.mtu_changed_callback = NULL;
+            break;
+
+        default:
+            /* Don't want to assert. New event callbacks may be added in the future */
+            return eBTStatusFail;
+    }
+
+    return eBTStatusSuccess;
+}
+
+/*******************************************************************************
+ * Unity fixtures
+ ******************************************************************************/
+void setUp( void )
+{
+    /* There's several functions which will be called frequently, and need stubs */
+    IotSemaphore_Create_Stub( IotSemaphore_Create_Callback );
+    IotSemaphore_Post_Stub( IotSemaphore_Post_Callback );
+    IotSemaphore_Destroy_Stub( IotSemaphore_Destroy_Callback );
+    IotSemaphore_TimedWait_Stub( IotSemaphore_TimedWait_Callback );
+
+    IotBle_CreateService_Stub( IotBle_CreateService_Callback );
+    IotBle_DeleteService_Stub( IotBle_DeleteService_Callback );
+
+    pvPortMalloc_Stub( pvPortMalloc_Callback );
+    vPortFree_Stub( vPortFree_Callback );
+
+    IotBle_RegisterEventCb_Stub( IotBle_RegisterEventCb_Callback );
+    IotBle_UnRegisterEventCb_Stub( IotBle_UnregisterEventCb_Callback );
+
+    IotLog_Generic_Ignore();
+}
+
+/* called before each testcase */
+void tearDown( void )
+{
+}
+
+/* called at the beginning of the whole suite */
+void suiteSetUp()
+{
+    n_dummy_callback_calls = 0;
+}
+
+/* called at the end of the whole suite */
+int suiteTearDown( int numFailures )
+{
+    return( numFailures > 0 );
+}
+
+
+/*******************************************************************************
+ * IotBleDataTransfer_Init
+ ******************************************************************************/
+
+/*
+ * Happy path
+ */
+void test_IotBleDataTransfer_Init_HappyPath( void )
+{
+    bool ret = false;
+
+    ret = IotBleDataTransfer_Init();
+    TEST_ASSERT_TRUE_MESSAGE( ret, "Failed to initialize data" );
+}
+
+/*
+ * Force various failures of callback registration within _registerCallbacks
+ */
```

Comment:
Can you give more details for the comments
possibly doxygen style?

Reply:
Addressed in later commit